sdk/appconfiguration/spring-cloud-azure-appconfiguration-config-web/src/main/java/com/microsoft/azure/spring/cloud/config/web/AppConfigurationWebAutoConfiguration.java

```diff
@@ -0,0 +1,29 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.microsoft.azure.spring.cloud.config.web;
+
+import org.springframework.boot.autoconfigure.condition.ConditionalOnBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.cloud.endpoint.RefreshEndpoint;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+import com.microsoft.azure.spring.cloud.config.AppConfigurationRefresh;
+
+/**
+ * Sets up refresh methods based on dependencies.
+ */
+@Configuration
+@ConditionalOnBean(AppConfigurationRefresh.class)
+public class AppConfigurationWebAutoConfiguration {
```

Comment:
this pr seems to add many spring related types. is this intentional?

Reply:
Yes, this was intentional as many Spring classes were missing Javadocs which resulted in Java 17 Javadocs to fail as it requires the missing Javadocs

Reply:
@alzimmermsft this is all incorrect, at least as I see it. This is undoing https://github.com/Azure/azure-sdk-for-java/pull/28970, which was intentional.